core/trino-main/src/test/java/io/trino/server/security/oauth2/TestOAuth2Security.java

```diff
@@ -0,0 +1,146 @@
+            try {
+                hydraIdP.close();
+            }
+            catch (Exception ex) {
+                throw new RuntimeException("Failed to stop Hydra Idp", ex);
+            }
+        });
+        hydraIdP.start();
+
+        String idpUrl = "https://localhost:" + hydraIdP.getAuthPort();
+
+        server = closer.register(
+                TestingTrinoServer.builder()
+                        .setCoordinator(true)
+                        .setAdditionalModule(binder -> jaxrsBinder(binder).bind(TestResourceSecurity.TestResource.class))
+                        .setProperties(ImmutableMap.<String, String>builder()
+                                .put("http-server.authentication.type", "oauth2")
+                                .put("http-server.https.enabled", "true")
+                                .put("http-server.https.keystore.path", Resources.getResource("cert/localhost.pem").getPath())
+                                .put("http-server.https.keystore.key", "")
+                                .put("http-server.authentication.oauth2.issuer", "https://localhost:4444/")
+                                .put("http-server.authentication.oauth2.auth-url", idpUrl + "/oauth2/auth")
+                                .put("http-server.authentication.oauth2.token-url", idpUrl + "/oauth2/token")
+                                .put("http-server.authentication.oauth2.jwks-url", idpUrl + "/.well-known/jwks.json")
+                                .put("http-server.authentication.oauth2.client-id", TRINO_CLIENT_ID)
+                                .put("http-server.authentication.oauth2.client-secret", TRINO_CLIENT_SECRET)
+                                .put("http-server.authentication.oauth2.user-mapping.pattern", "(.*)@(.*)")
+                                .put("http-server.authentication.oauth2.additional-audiences", TRINO_AUDIENCE)
+                                .put("oauth2-jwk.http-client.trust-store-path", Resources.getResource("cert/localhost.pem").getPath())
+                                .build())
+                        .build());
+
+        hydraIdP.createClient(
+                TRINO_CLIENT_ID,
+                TRINO_CLIENT_SECRET,
+                CLIENT_SECRET_BASIC,
+                ImmutableList.of(TRINO_AUDIENCE),
+                server.getHttpsBaseUrl() + "/oauth2/callback");
+
+        OkHttpClient.Builder clientBuilder = new OkHttpClient.Builder()
+                .followRedirects(false);
+        setupSsl(
+                clientBuilder,
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.of(LOCALHOST_KEYSTORE),
+                Optional.empty(),
+                Optional.empty());
+        client = clientBuilder.build();
+    }
+
+    @AfterClass
+    public void cleanup()
+            throws IOException
+    {
+        closer.close();
+    }
+
+    @Test
+    public void testAuthenticatorUserMapping()
```

Comment:
Can we add a test to existing test classe?

Reply:
It will make it a little more blurry. I will try.

Reply:
Done